numpy/core/src/umath/scalarmath.c.src

```diff
@@ -1564,6 +1564,13 @@ static PyObject*
 }
 /**end repeat**/
 
+#if PY_VERSION_HEX >= 0x03050000
+static PyObject *
+scalar_matmul(PyObject *self, PyObject *other)
+{
+    return NULL;
```

Comment:
Doesn't this need to set an error?

Reply:
It was never tested. Adding test, and simply removing the function to fix the test failure